### PR TITLE
update database

### PR DESCRIPTION
**Pull request by ridhishjain**
#### Changes: 
* changed the table name `user` to `users`
* removed `logged_in` columns from `users` table as they are of no use
* created table `notifications` and `user_notifications`
* created table `groups` and `user_groups`
* updated `database.sql`

### CODE REVIEW

File: server/database.sql [hunk 1]
```diff
@@ -18,28 +18,103 @@ USE `cl_judge`;
...
+
+DROP TABLE IF EXISTS `notifications`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `notifications` (
+  `id` varchar(45) NOT NULL,
+  `heading` varchar(110) NOT NULL,
+  `description` varchar(110) DEFAULT NULL,
+  `created_at` varchar(110) NOT NULL,
+  `link` varchar(110) NOT NULL,
```
Comment (cjchirag7) [inline server/database.sql:50]: link should be optional

File: server/database.sql [hunk 2]
```diff
@@ -18,28 +18,103 @@ USE `cl_judge`;
...
+
+DROP TABLE IF EXISTS `users`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `users` (
   `username` varchar(45) NOT NULL,
   `secret` longtext NOT NULL,
   `full_name` varchar(45) NOT NULL,
   `admission_number` varchar(45) NOT NULL,
-  `is_admin` tinyint DEFAULT NULL,
```
Comment (cjchirag7) [inline server/database.sql:32]: This should not be removed

**Review by cjchirag7 — CHANGES_REQUESTED** [review]
Let's make a change for better readability of code - group admins can be called `moderators`
And do note that admins users will only have the permission to create a new group.

File: server/database.sql [hunk 3]
```diff
@@ -18,28 +18,104 @@ USE `cl_judge`;
...
+
+DROP TABLE IF EXISTS `notifications`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `notifications` (
+  `id` varchar(45) NOT NULL,
+  `heading` varchar(110) NOT NULL,
+  `description` varchar(110) DEFAULT NULL,
+  `created_at` varchar(110) NOT NULL,
+  `link` varchar(110) DEFAULT NULL,
```
Comment (ridhishjain) [inline server/database.sql:50]: link is made optional

File: server/database.sql [hunk 4]
```diff
@@ -18,28 +18,104 @@ USE `cl_judge`;
...
+--
+
+DROP TABLE IF EXISTS `user_groups`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `user_groups` (
+  `id` varchar(45) NOT NULL,
+  `username` varchar(45) NOT NULL,
+  `group_id` varchar(45) NOT NULL,
+  `is_group_moderator` tinyint NOT NULL,
```
Comment (ridhishjain) [inline server/database.sql:67]: `is_admin` changed to `is_group_moderator`

File: server/database.sql [hunk 5]
```diff
@@ -18,28 +18,104 @@ USE `cl_judge`;
...
+CREATE TABLE `users` (
   `username` varchar(45) NOT NULL,
   `secret` longtext NOT NULL,
   `full_name` varchar(45) NOT NULL,
   `admission_number` varchar(45) NOT NULL,
-  `is_admin` tinyint DEFAULT NULL,
   `email` varchar(45) NOT NULL,
   `mobile` varchar(45) NOT NULL,
-  `logged_in` tinyint DEFAULT NULL,
+  `is_admin` tinyint DEFAULT NULL,
```
Comment (ridhishjain) [inline server/database.sql:111]: `is_admin` in users table for super-users like profs
By default null...

A separate API is to be built for making `is_admin = true`....the person signing up is required to upload some id proof or something (of being a prof), which will be verified by maintainers and then the person will be can be enabled as admin

Comment (cjchirag7) [reply server/database.sql:111]: We can do it mannually.

File: server/database.sql (hunk 4 above)
Comment (cjchirag7) [reply server/database.sql:67]: great!

**Review by cjchirag7 — APPROVED** [review]

**Pull request merged**